pallets/dkg-proposals/src/lib.rs

```diff
@@ -431,6 +431,27 @@ pub mod pallet {
 	}
 }
 
+impl<T: Config> ProposalsTrait<T::Proposal> for Pallet<T> {
+	/// Checks if specified proposal exists in the queue
+	fn proposal_exists(chain_id: u64, nonce: DepositNonce, prop: T::Proposal) -> bool {
+		let src_id = match T::ChainId::try_from(chain_id) {
+			Ok(v) => v,
+			Err(_) => return false,
+		};
+		return Votes::<T>::contains_key(src_id, (nonce, prop))
+	}
+
+	/// Removes specified proposal from the queue
+	fn remove_proposal(chain_id: u64, nonce: DepositNonce, prop: T::Proposal) -> bool {
```

Comment:
Why not make `chain_id` the proper type? 
```suggestion
	fn remove_proposal(chain_id: T::ChainId, nonce: DepositNonce, prop: T::Proposal) -> bool {
```